scripts/test

```diff
@@ -7,21 +7,25 @@ if [[ -n "${STAC4S_DEBUG}" ]]; then
 fi
 
 function usage() {
-    echo -ne \
-         "Usage: $(basename "${0}")
-Run linters and tests.
+    echo -n \
+        "Usage: $(basename "$0")
+Lint Scala source code and execute tests.
 "
 }
 
-
-if [ "${BASH_SOURCE[0]}" = "${0}" ]; then
-    case "${1}" in
-        --help)
-            usage
-            ;;
-        *)
-            echo -e "\e[32m[stac4s] Executing Scala Tests and Linting\e[0m"
-            ./sbt ";scalafix --check;scalafmtCheck;scalafmtSbtCheck;scapegoat;undeclaredCompileDependenciesTest;unusedCompileDependenciesTest;test"
-            ;;
-    esac
+if [[ "${BASH_SOURCE[0]}" == "${0}" ]]; then
+    if [[ "${1:-}" == "--help" ]]; then
+        usage
+    else
+        echo "Linting Scala source code and executing tests"
```

Comment:
I decided not to carry over the color codes because they weren't working on my system:

```bash
maiden:~ rbreslow$ echo -e "\e[32m[stac4s] Executing Scala Tests and Linting\e[0m"
\e[32m[stac4s] Executing Scala Tests and Linting\e[0m
```

Also, for consistency, we weren't using them in `cipublish`.

If we were to include color codes, I would suggest an [alternative approach](https://stackoverflow.com/a/5413029) that is portable (with `printf`) and provides more readable code:

```diff
diff --git a/scripts/test b/scripts/test
index 8a4de5f..816a1e7 100755
--- a/scripts/test
+++ b/scripts/test
@@ -13,11 +13,16 @@ Lint Scala source code and execute tests.
 "
 }

+function success() {
+    printf "$(tput setaf 2)%s$(tput sgr0)\n" $1
+}
+
 if [[ "${BASH_SOURCE[0]}" == "${0}" ]]; then
     if [[ "${1:-}" == "--help" ]]; then
         usage
     else
-        echo "Linting Scala source code and executing tests"
+        success "Linting Scala source code and executing tests"
+
         ./sbt ";\
             scalafix --check; \
             scalafmtCheck; \
```


I would also consider whether the value the color codes add to the script outweighs what they subtract in terms of complexity and decreased readability. If it is something we want to standardize on, which is totally cool, it may be worth extending the STRTA ADR so we can get every repository on the same page going forward.

Reply:
Cool, I like your change -- in general I find the colors to be helpful in parsing the output of our STRTA commands (what's the command vs what's log output from processes the command invokes to be helpful). As to whether the increase in complexity for STRTA outweighs its usefulness - I would say that once written our STRTA commands aren't edited that much, but we run them often so it seems like a fair tradeoff.

Nevertheless, I don't think this change should hold up the PR since it's more a matter of preference and the original edit didn't seem related to the goal of publishing artifacts.

Reply:
>in general I find the colors to be helpful in parsing the output of our STRTA commands (what's the command vs what's log output from processes the command invokes to be helpful). 

Agreed. It's definitely helpful.

>As to whether the increase in complexity for STRTA outweighs its usefulness - I would say that once written our STRTA commands aren't edited that much, but we run them often so it seems like a fair tradeoff.

The thing in my mind is, if we replicate a `success` or `log` function across all the STRTA, there's more to parse and possibly distract when reading each file. Embedding the color codes in the `printf` statement introduces a different set of readability concerns. I agree with your perspective though, if we standardize on this, it might eventually become background noise. 

At a higher level, it may be worth revisiting the STRTA ADR, because many projects have been deviating from the standards and that document should be updated for how we're developing apps today (Vagrant seems to be disused, etc).